### PR TITLE
Improve error messages in tests

### PR DESCRIPTION
**Pull request by ChrisJefferson**
This deals with GAP allowing `> ` in the middle of tests, and appending the line into the input of the test. While #3068 is not fixed by this, it does make the error (in my opinion) much easier to understand.

Also, this overhauls the errors produced by tests to give a line number, and add some basic tests (which have to be in testspecial, because they involve test files which fail).

### CODE REVIEW

File: tst/testspecial/broken-test.g.out [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+ ); at GAPROOT/lib/test.gi:124 called from
+ParseTestInput( full, opts.ignoreComments 
+ ) at GAPROOT/lib/test.gi:558 called from
+<function "Test">( <arguments> )
+ called from read-eval loop at *stdin*:2
+type 'quit;' to quit to outer loop
+brk> quit;
+gap> Test("broken-test-3.tst");
+Error, Invalid data at line 5 of test file at GAPROOT/lib/test.gi:24 called from
+testError( "Invalid data" ); at GAPROOT/lib/test.gi:130 called from
```
Comment (fingolfin) [inline tst/testspecial/broken-test.g.out]: Apparently this line causes a diff; perhaps you need to run the regenerate script once again?

File: lib/test.gi [hunk 2]
```diff
@@ -12,14 +12,17 @@
...
+  local lines, inp, pos, outp, ign, commands, i, skipstate, checkifelseendif, foundcmd, testError;
   lines := SplitString(str, "\n", "");
   inp := [];
   pos := [];
   outp := [];
   ign := [];
   commands := [];
   i := 1;
+  testError := function(s)
+    ErrorNoReturn(Concatenation(s, " at line ", String(i)), " of test file");
```
Comment (fingolfin) [inline lib/test.gi]:
```suggestion
    ErrorNoReturn(s, " at line ", i, " of test file");
```

Comment (fingolfin) [reply lib/test.gi]: This is still a bit unsatisfactory, it'd be nice if the name of the test file also appeared. Could we simply pass `fnam` to `ParseTestInput`, and then do something similar to what `Test` does, say,
```gap
             if IsStream(fnam) then
               Print(s, " in test stream, line ", line, ":\n");
             else
               Print(s, "in ", fnam, ":", line, "\n");
             fi;
```

File: tst/testspecial/broken-test.g.out (hunk 1 above)
Comment (ChrisJefferson) [reply tst/testspecial/broken-test.g.out]: Something weird is going on, I am getting the test working on my machine. I tried making the line length longer and that didn't fix it either... investigating

Comment (ChrisJefferson) [reply tst/testspecial/broken-test.g.out]: I have no idea why this isn't working. I checked the screen width isn't getting set too small. Any suggestions welcome.

**Comment by fingolfin** [comment]
That's odd. If I take your PR, remove your `TEMPCOMMITFORTESTING` commit, and run `make && cd tst/testspecial && GAPDIR= ./regenerate_tests.sh` I do get a diff matching what we see on Travis:
```patch
diff --git a/tst/testspecial/broken-test.g.out b/tst/testspecial/broken-test.g.out
index b624c7853..2f3f723a3 100644
--- a/tst/testspecial/broken-test.g.out
+++ b/tst/testspecial/broken-test.g.out
@@ -20,7 +20,8 @@ type 'quit;' to quit to outer loop
 brk> quit;
 gap> Test("broken-test-3.tst");
 Error, Invalid data at broken-test-3.tst:5 at GAPROOT/lib/test.gi:27 called from
-testError( "Invalid data" ); at GAPROOT/lib/test.gi:134 called from
+testError( "Invalid data" 
+ ); at GAPROOT/lib/test.gi:134 called from
 ParseTestInput( full, opts.ignoreComments, fnam 
  ) at GAPROOT/lib/test.gi:562 called from
 <function "Test">( <arguments> )
```

**Comment by ChrisJefferson** [comment]
Now, hopefully, all fine and tests passing.

File: tst/testspecial/broken-test.g.out [hunk 3]
```diff
@@ -0,0 +1,54 @@
+gap> Test("broken-test-1.tst", rec(width := 800));
+Error, Cannot put a '> ' continuation in the middle of test output at broken-test-1.tst:3 at GAPROOT/lib/test.gi:27 called from
```
[the remark this answers is not in the document]
Comment (ChrisJefferson) [reply tst/testspecial/broken-test.g.out]: I overhauled the errors in another way, to also make them more consistent with each other

**Pull request merged**